client/src/services/api/hooks/codeReview/codeReviewUtils.ts

```diff
@@ -45,5 +50,83 @@ export class CodeReviewMetricsUtils {
       pullRequests,
       errorCount: data.errorCount,
     };
-  };
+  }
+
+  static getFilteredPullRequestsByReviewers(
+    data: Omit<IFetchedRawCodeReviewResponse, "count" | "filteredCount"> | undefined,
+    filters: IContextClientFilterSquad[],
+  ): IFetchedCodeReviewResponse | undefined {
+    if (!data) {
+      return;
+    }
+
+    const { reviewersMap } = this.#getSquadsUserIdsMap(filters);
+
+    const pullRequests = data.pullRequests.map((pullRequest) => {
+      const filteredVotesTimeline = this.#filterVotesTimelineByReviewers(pullRequest.votesTimeline, reviewersMap);
+      const votes = this.#getPullRequestVotes(filteredVotesTimeline);
+
+      const filteredVotesHistoryTimeline = this.#filterVotesTimelineByReviewers(
+        pullRequest.votesHistoryTimeline,
+        reviewersMap,
+      );
+      const votesHistory = this.#getPullRequestVotesHistory(filteredVotesHistoryTimeline);
+
+      return {
+        ...pullRequest,
+        votes,
+        votesHistory,
+        votesTimeline: filteredVotesTimeline,
+        votesHistoryTimeline: filteredVotesHistoryTimeline,
+      };
+    });
+
+    return {
+      pullRequests,
+      errorCount: data.errorCount,
+    };
+  }
+
+  static #filterVotesTimelineByReviewers(
+    votesTimeline: IFetchedPullRequestVotesTimeline[],
+    reviewersMap: Map<string, boolean>,
+  ) {
+    if (!reviewersMap.size) {
+      return votesTimeline;
+    }
+
+    return votesTimeline.filter(({ id }) => reviewersMap.has(id));
+  }
+
+  static #getPullRequestVotes(votesCycle: IFetchedPullRequestVotesTimeline[]) {
+    const votesResults = {
+      [VOTES.APPROVED]: 0,
+      [VOTES.APPROVED_WITH_SUGGESTIONS]: 0,
+      [VOTES.WAIT_FOR_AUTHOR]: 0,
+      [VOTES.REJECTED]: 0,
+      [VOTES.NO_VOTE]: 0,
+    };
+
+    votesCycle.forEach(({ vote }) => {
+      votesResults[vote]++;
+    });
+
+    return votesResults;
+  }
+
+  static #getPullRequestVotesHistory(votesCycle: IFetchedPullRequestVotesTimeline[]) {
```

Comment:
Can we reuse this function `getPullRequestVotes`?

Reply:
Sure, Sathish, for now we can reuse but when we when we create separate interface for votesHistory (Omit<IPullRequestVotes, "noVote">, we can create separate functions.